tests/e2e/txgen/txgen.go

```diff
@@ -0,0 +1,204 @@
+	coreMemSig "github.com/oasisprotocol/oasis-core/go/common/crypto/signature/signers/memory"
+	cmnGrpc "github.com/oasisprotocol/oasis-core/go/common/grpc"
+	"github.com/oasisprotocol/oasis-core/go/common/quantity"
+
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/client"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature/ed25519"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature/secp256k1"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/modules/accounts"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+)
+
+const highGasAmount = 1000000
+
+// AccountType is the type of account to create.
+type AccountType uint8
+
+const (
+	AccountEd25519 AccountType = 0
+	AccountSecp256k1 AccountType = 1
+	AccountTypeMax = AccountSecp256k1
+)
+
+func (at AccountType) String() string {
+	return [...]string{"ed25519", "secp256k1"}[at]
+}
+
+// NewClient creates a new runtime client.
+func NewClient(clientNodeUnixSocketPath string, runtimeID common.Namespace) (client.RuntimeClient, error) {
+	conn, err := grpc.Dial(clientNodeUnixSocketPath,
+		grpc.WithDefaultCallOptions(grpc.ForceCodec(&cmnGrpc.CBORCodec{})),
+		grpc.WithInsecure(),
+		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+			return net.DialTimeout("unix", addr, timeout)
+		}))
+	if err != nil {
+		return nil, err
+	}
+	return client.New(conn, runtimeID), nil
+}
+
+// GetChainContext returns the chain context.
+func GetChainContext(ctx context.Context, rtc client.RuntimeClient) (signature.Context, error) {
+	info, err := rtc.GetInfo(ctx)
+	if err != nil {
+		return "", err
+	}
+	return info.ChainContext, nil
+}
+
+// EstimateGas estimates the amount of gas the transaction will use.
+// Returns modified transaction that has just the right amount of gas.
+func EstimateGas(ctx context.Context, rtc client.RuntimeClient, tx types.Transaction) types.Transaction {
+	var gas uint64
+	oldGas := tx.AuthInfo.Fee.Gas
+	// Set the starting gas to something high, so we don't run out.
+	tx.AuthInfo.Fee.Gas = highGasAmount
+	// Estimate gas usage.
+	if err := rtc.Query(ctx, client.RoundLatest, "core.EstimateGas", tx, &gas); err != nil {
+		tx.AuthInfo.Fee.Gas = oldGas
+		return tx
+	}
+	// Specify only as much gas as was estimated.
+	tx.AuthInfo.Fee.Gas = gas
+	return tx
+}
+
+// SignAndSubmitTx signs and submits the given transaction.
+// Gas estimation is done automatically.
+func SignAndSubmitTx(ctx context.Context, rtc client.RuntimeClient, signer signature.Signer, tx types.Transaction) error {
+	// Get chain context.
+	chainCtx, err := GetChainContext(ctx, rtc)
+	if err != nil {
+		return err
+	}
+
+	// Get current nonce for the signer's account.
+	ac := accounts.NewV1(rtc)
+	nonce, err := ac.Nonce(ctx, client.RoundLatest, types.NewAddress(signer.Public()))
+	if err != nil {
+		return err
+	}
+	tx.AppendSignerInfo(signer.Public(), nonce)
```

Comment:
we're changing the signature of AppendSignerInfo to be more general. could you replace the new calls with AppendAuthSignature